alacritty_terminal/src/window.rs

```diff
@@ -409,6 +425,39 @@ impl Window {
     }
 }
 
+#[cfg(not(any(target_os = "macos", windows)))]
+fn x_embed_window(window: &GlutinWindow, parent_id: u64) {
+    let (xlib_display, xlib_window) = match (window.get_xlib_display(), window.get_xlib_window()) {
+        (Some(display), Some(window)) => (display, window),
+        _ => return,
+    };
+
+    let xlib = Xlib::open().expect("get xlib");
```

Comment:
If there's a way to get the `xconn.xlib` that winit already has? That way, ideally, we don't need to check for this error again here.

Reply:
I think we could use [get_xcb_connection](https://docs.rs/winit/0.19.3/winit/os/unix/trait.WindowExt.html#tymethod.get_xcb_connection) and [get_xlib_display](https://docs.rs/winit/0.19.3/winit/os/unix/trait.WindowExt.html#tymethod.get_xlib_display) to construct an [XCB Connection](http://rtbo.github.io/rust-xcb/xcb/base/struct.Connection.html) and then use [reparent_window](http://rtbo.github.io/rust-xcb/xcb/xproto/fn.reparent_window.html) to embed the [winit window](https://docs.rs/winit/0.19.3/winit/os/unix/trait.WindowExt.html#tymethod.get_xlib_window).